game-core/src/main/java/games/strategy/triplea/delegate/TechAdvance.java

```diff
@@ -185,7 +179,7 @@ public static void createDefaultTechAdvances(final GameData data) {
    *     instance.
    */
   public static TechAdvance findDefinedAdvanceAndCreateAdvance(
-      final String technologyName, final GameState data) {
```

Comment:
This is actually an important warning that was pointed out by intelliJ.
The reflection only searches for a constructor that accepts an instance of `GameData`, which is to be fair currently the only implementing class of the `GameState` interface, but if there was another implementation this method would throw an exception